cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexUpdaterTest.java

```diff
@@ -1041,6 +1041,114 @@ public Void call(SQLDatabase db) throws Exception {
 
     }
 
+    @Test
+    public void indexUpdatesPersistFromCreation() throws Exception {
+        long exepctedSequence = 0l;
+        try {
+            DocumentRevision rev = new DocumentRevision("mike12");
+            // body content: { "name" : "mike",  "age" : 12, "pet" : "cat" }
+            Map<String, Object> bodyMap = new HashMap<String, Object>();
+            bodyMap.put("name", "mike");
+            bodyMap.put("age", 12);
+            bodyMap.put("pet", "cat");
+            rev.setBody(DocumentBodyFactory.create(bodyMap));
+            ds.createDocumentFromRevision(rev);
+
+            createIndex("testIndex", Collections.singletonList((Object) "name"));
+            exepctedSequence = getIndexSequenceNumber("testIndex");
+            Assert.assertEquals("The expected sequence should be 1 for 1 document created.", 1,
+                    exepctedSequence);
+        } finally {
+            im.close();
+        }
+        // Now get a new IndexManager and queue and update the test fields accordingly
```

Comment:
Replaced with the hopefully more clear 
`// Get a new IndexManager instance and extract its queue`
in 529262a